examples/demo_threading/src/lib.rs

```diff
@@ -0,0 +1,335 @@
+    };
+    use futures::{executor::block_on, stream::StreamExt};
+    use futures_timer::Delay;
+    use std::{
+        collections::HashMap,
+        sync::mpsc::{sync_channel, Receiver, SyncSender},
+        thread::JoinHandle,
+        time::{Duration, SystemTime},
+    };
+    use uuid::Uuid;
+
+    pub struct Data {
+        average_use: f64,
+        sensors: u32,
+        total_use: f64,
+    }
+
+    impl Default for Data {
+        fn default() -> Self {
+            Self {
+                average_use: 0.0,
+                sensors: 0,
+                total_use: 0.0,
+            }
+        }
+    }
+
+    struct RustObj {
+        qt_rx: Receiver<Data>,
+        qt_tx: SyncSender<Data>,
+        join_handle_network: Option<JoinHandle<()>>,
+        join_handle_timeout: Option<JoinHandle<()>>,
+        join_handle_update: Option<JoinHandle<()>>,
+    }
+
+    impl Default for RustObj {
+        fn default() -> Self {
+            let (qt_tx, qt_rx) = sync_channel(super::CHANNEL_QT_COUNT);
+            Self {
+                qt_rx,
+                qt_tx,
+                join_handle_network: None,
+                join_handle_timeout: None,
+                join_handle_update: None,
+            }
+        }
+    }
+
+    impl RustObj {
+        async fn handle_connection(mut stream: TcpStream, network_tx: SyncSender<NetworkChannel>) {
+            let mut buf = vec![0u8; 1024];
+            let _ = stream.read(&mut buf).await.unwrap();
+            let trimmed = std::str::from_utf8(&buf)
+                .unwrap()
+                .trim_matches(|c| c == ' ' || c == '\n' || c == '\r' || c == '\0');
+
+            let response: Response = match serde_json::from_str::<Request>(trimmed) {
+                Ok(request) => {
+                    match request.command {
+                        RequestCommand::Power { value } => {
+                            // Validate that our power is within the expected range
+                            if (0.0..=super::SENSOR_MAXIMUM_POWER).contains(&value) {
+                                network_tx
+                                    .try_send(NetworkChannel::Power {
+                                        uuid: request.uuid,
+                                        value,
+                                    })
+                                    .into()
+                            } else {
+                                Status::ErrorInvalidPower.into()
+                            }
+                        }
+                        RequestCommand::Disconnect => network_tx
+                            .try_send(NetworkChannel::Disconnect { uuid: request.uuid })
+                            .into(),
+                    }
+                }
+                Err(_) => Status::ErrorInvalidRequest.into(),
+            };
+
+            stream
+                .write(serde_json::to_string(&response).unwrap().as_bytes())
+                .await
+                .unwrap();
```

Comment:
Again, no unwrap here, please, just using `.ok()` here should be fine.

Reply:
Done